### PR TITLE
ci: setup sonarcloud

### PR DESCRIPTION
**Pull request by andrueastman**
Fixes https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/issues/904
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/pull/905)

### CODE REVIEW

File: .github/workflows/sonarcloud.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        id: checksecret_job
+        run: |
+          echo "is_SONAR_TOKEN_set=${{ env.SONAR_TOKEN != '' }}" >> $GITHUB_OUTPUT
+  build:
+    needs: [checksecret]
+    if: needs.checksecret.outputs.is_SONAR_TOKEN_set == 'true'
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up JDK 17
```
Comment (baywet) [inline .github/workflows/sonarcloud.yml:34]: I believe we could go straight to 21 to buy more runway in terms of EOL.

File: .github/workflows/sonarcloud.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        uses: actions/setup-java@v4
+        with:
+          distribution: "adopt"
+          java-version: 17
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v4
+        with: # At the moment the scanner requires dotnet 5 https://www.nuget.org/packages/dotnet-sonarscanner
+          dotnet-version: |
```
Comment (baywet) [inline .github/workflows/sonarcloud.yml:42]: This is not true anymore. https://www.nuget.org/packages/dotnet-sonarscanner/8.0.1#supportedframeworks-body-tab
We should be able to bump to net 8

File: .github/workflows/sonarcloud.yml (hunk 1 above)
Comment (andrueastman) [reply .github/workflows/sonarcloud.yml]: Confirmed in latest update.

**Review by baywet — APPROVED** [review]
Thank you for making the changes!

**Pull request merged**